client/lib/fifo/fifo_test.go

```diff
@@ -50,13 +48,14 @@ func TestFIFO(t *testing.T) {
 	go func() {
 		defer wait.Done()
 
-		var err error
-		reader, err = readerOpenFn()
 		assert.NoError(t, err)
 		if err != nil {
 			return
 		}
 
+		reader, err = openFn()
+		require.NoError(err)
```

Comment:
I think there a couple of issues here:
* In the goroutine, we cannot use `require.NoError` effectively, so must use `assert` instead.
* We want to have a goroutine specific error variable rather than reuse the err variable in multiple goroutine and have race conditions.  With this change, `assert.NoError(t, err)` currently assert on errors unrelated to the goroutine.

Any particular issue here you tried to address with this change?

Reply:
Nope - This is a hangover from another change, I'll revert it.